internal/network/command_dns_record_list.go

```diff
@@ -0,0 +1,55 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/output"
+)
+
+func (c *command) newDnsRecordListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List DNS records.",
+		Args:  cobra.NoArgs,
+		RunE:  c.dnsRecordList,
+	}
+
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
```

Comment:
There are a few parameters for `List`, not sure if the bug in the SDK for `MultipleSearchFilter` is fixed...

Reply:
I added four flags for filtering the dns search results based on the methods for `ApiListNetworkingV1DnsRecordsRequest`.

Reply:
If there's an issue with using multiple filters then I can just restrict it in the CLI so that only one of the flags can be specified.